coingecko/errors.ts

```diff
@@ -1,25 +0,0 @@
-export enum ErrorType {
```

Comment:
please don't use enums, `{} as const` is the real way to go due to typescript internal issues with enums

Reply:
I don't see the reason to do it instead of enum for 4 custom error codes. It whould be great if you eleborate on that a bit more

Reply:
https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh